pubsublite/pubsublite.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package pubsublite
+
+import (
+	"context"
+
+	"google.golang.org/api/option"
+
+	vkit "cloud.google.com/go/pubsublite/apiv1"
+	pb "google.golang.org/genproto/googleapis/cloud/pubsublite/v1"
+)
+
+// Client provides admin operations for Google Pub/Sub Lite resources within a
+// Google Cloud region. A Client may be shared by multiple goroutines.
+type Client struct {
```

Comment:
Hmm, I might leave this as Client (and changed NewAdminClient to NewClient) to match pubsub.Client. One of our goals was to make the pubsublite interface as consistent as possible with pubsub. Given that we've had to make quite a few deviations already, we could live with this one. (If starting from scratch, I would actually call this AdminClient too.)

Reply:
@hongalex, do you have a strong preference?

Reply:
Considering we'll using separate publisher/subscriber clients for Lite, I think the deviation makes sense. Keeping it the same might confuse new users anyway since Pub/Sub uses the generic `Client` struct for everything, something that Lite won't do.

Reply:
Seems no one is particularly attached to `Client` :)
I've renamed to AdminClient and the file to admin.go.